File: problem/Robot_Origin/judgeCircle.js
/**
 * @param {string} moves
 * @return {boolean}
 */
const judgeCircle = moves => {
    if (moves.length & 1) return false;
    const freq = { 'U': 0, 'D': 0, 'L': 0, 'R': 0 };

    for (const c of moves)
        freq[c]++;

    return freq.U === freq.D && freq.L === freq.R;
};

File: problem/Robot_Origin/judgeCircle.cpp
class Solution
{
public:
    bool judgeCircle(string moves)
    {
        if (moves.length() & 1)
            return false;
        int u = 0, d = 0, l = 0, r = 0;

        for (char c : moves)
        {
            u += c == 'U';
            d += c == 'D';
            l += c == 'L';
            r += c == 'R';
        }

        return u == d && l == r;
    }
};